engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/gateway/ExclusiveGatewayTest.java

```diff
@@ -365,4 +365,55 @@ public void shouldResolveIncidentsWhenTerminating() {
         .extracting(Record::getIntent)
         .containsExactly(IncidentIntent.CREATED, IncidentIntent.RESOLVED);
   }
+
+  @Test
+  public void shouldNotEvaluateConditionOfDefaultFlow() {
+    // given
+    final String processId = Strings.newRandomValidBpmnId();
+    final BpmnModelInstance processDefinition =
+        Bpmn.createExecutableProcess(processId)
+            .startEvent()
+            .exclusiveGateway("xor")
+            .sequenceFlowId("s1")
+            .conditionExpression("foo >=1 and foo < 5")
+            .endEvent("a")
+            .moveToLastGateway()
+            .sequenceFlowId("s2")
+            .conditionExpression("foo >= 5 and foo < 10")
+            .endEvent("b")
+            .moveToLastExclusiveGateway()
+            .defaultFlow()
+            .sequenceFlowId("s3")
+            .conditionExpression("foo >= 10")
+            .endEvent("c")
+            .done();
```

Comment:
❌ This test does not verify that the condition on the default flow is not evaluated

I suggest replacing your test case with two individual test cases:

1. Test that an end event is reached, even though there is a mistake in the default condition (`missing_var` is a reference to a variable that is not available in the process instance). This allows checking that the condition is not evaluated because if it were evaluated, an incident is raised, stopping the flow from reaching the end event.
<img width="409" alt="Screen Shot 2022-11-10 at 16 48 17" src="https://user-images.githubusercontent.com/3511026/201141260-b853e04a-d188-4cc0-be9a-d41d5825496d.png">

2. The same test as above, but with a non-default sequence flow present as well, that evaluates to `false` so that the default flow must be taken. We should still check that the condition on the default flow is not evaluated, and we can do that in the same way as in the first test.
<img width="553" alt="Screen Shot 2022-11-10 at 16 51 25" src="https://user-images.githubusercontent.com/3511026/201142102-b6364258-fe19-4ebe-9210-88c269bdedbe.png">

When you add these tests, you'll notice that we need to make a change to the ExclusiveGatewayProcessor as well to avoid evaluating the condition on the default flow.